### PR TITLE
fix(reports): validator_config, report state machine, working_timeout

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Fixes/Improvement to the existing implementation:
- marshmallow schema for `validator_config_json` `op` was called `operation`: **This needs a FE change**
- Introduces a new state called `GRACE` when alerts enter a grace period after a `SUCCESS`: **This needs a FE change**
- Add a column to `ReportSchedule` named `working_timeout` so that bogus reports do not enter a dead end state: **This need a FE change** (this new column can/should be set by the user when configuring a new Report/Alert)
- Fetches the `THUMBNAIL_SELENIUM_USER` using current worker scoped ORM session (not security manager session)
- `AsyncExecuteReportScheduleCommand` calls a state machine, more readable code and easier to reason and extend

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [X] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

@mihir174

### CODE REVIEW

File: superset/reports/commands/execute.py [hunk 1]
```diff
@@ -192,45 +200,173 @@ def _send(self, report_schedule: ReportSchedule) -> None:
...
+        self,
+        session: Session,
+        report_schedule: ReportSchedule,
+        scheduled_dttm: datetime,
+    ):
+        self._session = session
+        self._report_schedule = report_schedule
+        self._scheduled_dttm = scheduled_dttm
+
+    def run(self) -> None:
```
Comment (willbarrett) [inline superset/reports/commands/execute.py:334]: It looks like this will silently swallow not-found states?

Comment (dpgaspar) [reply superset/reports/commands/execute.py:335]: good point, yet it would be very unexpected. Addressed it

**Review by villebro — APPROVED** [review]
This looks great! Some minor proposals, but nothing blocking.

File: superset/reports/commands/alert.py [hunk 2]
```diff
@@ -70,7 +76,7 @@ def _validate_operator(self, rows: np.recarray) -> None:
             raise AlertQueryMultipleColumnsError(
                 _(
                     "Alert query returned more then one column. %s columns returned"
-                    % len(rows[0])
+                    % (len(rows[0]) - 1)
```
[the remark this answers is not in the document]
Comment (dpgaspar) [reply superset/reports/commands/alert.py:79]: Exactly that, I'll drop a comment

[the remark this answers is not in the document]
Comment (dpgaspar) [reply superset/reports/commands/alert.py:80]: done

**Review by willbarrett — APPROVED** [review]

File: superset/models/reports.py [hunk 3]
```diff
@@ -122,7 +123,10 @@ class ReportSchedule(Model, AuditMixinNullable):
 
     # Log retention
     log_retention = Column(Integer, default=90)
+    # (Alerts) After a success how long to wait for a new trigger (seconds)
     grace_period = Column(Integer, default=60 * 60 * 4)
+    # (Alerts/Reports) Unlock a possible stalled working state
+    working_timeout = Column(Integer, default=60 * 60 * 1)
```
[the remark this answers is not in the document]
Comment (dpgaspar) [reply superset/models/reports.py:129]: this default is actually imposed by SQLAlchemy only, server_default would place it at the db level. I confess I did not knew that :(

**Pull request merged**